platform/stm32f4xx/export/platform/i2c_bus.hpp

```diff
@@ -0,0 +1,648 @@
+#ifndef PLATFORM_I2C_BUS_HPP_
+#define PLATFORM_I2C_BUS_HPP_
+
+#include <stm32f4xx_i2c.h>
+#include <stm32f4xx_rcc.h>
+
+#include <platform/irq_manager.hpp>
+
+#include <common/i2c.hpp>
+
+#include <sys/types.h>
+
+namespace ecl {
+
+enum class i2c_mode {
+    POLL,
+    IRQ
+};
+
+template< i2c_device        dev,
+          i2c_mode          mode,
+          uint32_t          clock_speed,
+          uint16_t          operation_mode,
+          uint16_t          duty_cycle,
+          uint16_t          own_address1,
+          uint16_t          ack,
+          uint16_t          acknowledged_address>
+struct i2c_config
+{
+    static constexpr I2C_InitTypeDef m_init_obj = {
+            clock_speed,
```

Comment:
Use designators when initializing a struct, this will help avoid issues with initialization of wrong fields.